src/server/svr_func.c

```diff
@@ -5102,67 +5102,10 @@ check_and_run_jobs(struct prov_vnode_info * prov_vnode_info)
 		return;
 
 	rc = is_runnable(pjob, prov_vnode_info);
-	if (rc == 0) {
-		if (do_sync_mom_hookfiles || sync_mom_hookfiles_proc_running) {
-
-			/**
-			 * If mom hook files sync is in process then create
-			 * a time task where you perform this check again,
-			 * and start the job once it is done
-			 */
 
-			DBPRT(("%s: setting the time task as sync mom"
-				"hookfiles is not completed\n"
-				, __func__))
 
-			/* set a work task to run after 5 sec from now */ 
-			if (!set_task(WORK_Timed, time_now + 5, 
-				is_hooks_sync_done, pjob)){
-				log_err(errno, __func__,
-					"Unable to set task for is_hooks_sync_done; requeuing the job");
-				(void)force_reque(pjob);
-			}
-			return;
-		}
-
-		/*  accounting log about prov for job over */
-		set_job_ProvAcctRcd(pjob, time_now,
-			PROVISIONING_SUCCESS);
-
-		/* log msg about prov for job over */
-		sprintf(log_buffer,
-			"Provisioning for Job %s succeeded, running job",
-			pjob->ji_qs.ji_jobid);
-		log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB,
-			LOG_INFO, pjob->ji_qs.ji_jobid,
-			log_buffer);
-
-		DBPRT(("%s: Jobid: %s about to run after prov success\n",
-			id,  pjob->ji_qs.ji_jobid))
-
-		/* now prov_vnode is stale, remove it */
-		if (pjob->ji_wattr[(int)JOB_ATR_prov_vnode].at_flags &
-			ATR_VFLAG_SET) {
-			job_attr_def[(int)JOB_ATR_prov_vnode].at_free(
-				&pjob->ji_wattr[(int)JOB_ATR_prov_vnode]);
-		}
-
-		/* Move the job to MOM */
-		pjob->ji_modified = 1;
-		if ((rc = svr_startjob(pjob, 0))!=0) {
-			DBPRT(("%s: Jobid: %s - startjob failed - rc:%d\n",
-				id, pjob->ji_qs.ji_jobid, rc))
-			free_nodes(pjob);
-			if (pjob->ji_qs.ji_svrflags &
-				JOB_SVFLG_SubJob) {
-				/* requeue subjob */
-				pjob->ji_qs.ji_substate =
-					JOB_SUBSTATE_RERUN3;
-				job_purge(pjob);
-			}
-		}
-		DBPRT(("%s: Jobid: %s, startjob returned: %d\n",
-			id, pjob->ji_qs.ji_jobid, rc))
+	if (rc == 0) {
+		set_task(WORK_Immed, 0, is_hooks_sync_done, pjob);
```

Comment:
why do you set the work task? why don't you just call the function is_hooks_sync_done here?